src/solvers/refinement/string_constraint_generator_format.cpp

```diff
@@ -486,9 +486,7 @@ utf16_constant_array_to_java(const array_exprt &arr, std::size_t length)
   for(std::size_t i=0; i<arr.operands().size() && i<length; i++)
   {
     PRECONDITION(arr.operands()[i].id()==ID_constant);
-    auto c = numeric_cast<unsigned>(to_constant_expr(arr.operands()[i]));
-    INVARIANT(c.has_value(), "constant should be convertible to unsigned");
-    out[i] = c.value();
+    out[i] = numeric_cast_v<unsigned>(to_constant_expr(arr.operands()[i]));
```

Comment:
The above `PRECONDITION` is rather redundant.

Reply:
I do like it as it clarifies the contract here well; will move up.